gamemodes/terrortown/gamemode/shared/sh_sprint.lua

```diff
@@ -85,24 +108,23 @@ hook.Add("Think", "TTT2PlayerSprinting", function()
 
 	for i = 1, #plys do
 		local ply = plys[i]
+		local wantsToMove = ply:KeyDown(IN_FORWARD) or ply:KeyDown(IN_BACK) or ply:KeyDown(IN_MOVERIGHT) or ply:KeyDown(IN_MOVELEFT)
 
-		if not ply.sprintTS then continue end
+		if ply.sprintProgress == 1 and (not ply.isSprinting or not wantsToMove) then continue end
+		if ply.sprintProgress == 0 and ply.isSprinting and wantsToMove then continue end
```

Comment:
So I need to release the key manually to be able to regenerate? Or will it start automatically? How about having just a slower sprint like in the current version?

Reply:
And will the player start regenerating even if he is moving? The stamina should just be consumed if the player sprints.

Reply:
1. You will regenerate stamina while standing still, regardless if you press the sprint bind or not.
2. The stamina is only ever consumed if the player is moving.
3. If you keep trying to sprint while moving, but have no stamina, the stamina will not start regenerating (I tested it and it would just feel kind of off if it does)
4. If you just walk around normally without trying to sprint, the stamina will regenerate, just like before.

Number 1 is the only thing that's different from before, except you can now also start sprinting by double tapping w.